src/main/kotlin/traffic_simulation/Vehicle.kt

```diff
@@ -1,6 +1,6 @@
 package traffic_simulation
 
-class Vehicle(val capacityFactor: Double){
+class Vehicle(val id : Int , val wannaDrive : Boolean , val capacityFactor: Double = 1.0){
```

Comment:
Wenn du nichts weiteres in `Vehicle` und eventuell auch anderswo änderst, wird `wannaDrive` nie verwendet. Ich denke hier musst du mit @jwustmann besprechen wie genau die Lösung aussehen soll.

Meine Meinung:

* `capacityFactor` kommt weg, weil es bisher nicht benötigt wird
* `Vehicle` bekommt eine Methode `wannaDrive()` die einen `Boolean` zurückgibt
* `RoadNetwork` ruft `wannaDrive()` auf und erhöht die benötigte Kapazität um 1, falls `true` zurückkommt